### ISSUE TITLE
'__in' lookups don't work with CharFilter

### ISSUE THREAD

**Issue by jameswyse** [bennullgraham/filternaut#2]
`CharFilter` doesn't seem to work with `__in` lookups.. the array gets converted to a string when `CharFilter` calls `clean`, so `field__in=['one', 'two']` becomes `field__in="['one', 'two']"` 😭 

Is this a bug? Either way I switched to using `Filter` and it's working great :)

Here's an example:

### Filters
```python
filters = (
    CharFilter('foo__bar', lookups=['exact', 'in']),
    Filter('bar__baz', lookups=['exact', 'in']),
)
```

### Using `CharFilter`:
```python
filters.parse(MultiValueDict({
    'foo__bar__in': ['one', 'two', 'three']
}))

# Results:
filters.valid == True
filters.Q == <Q: (AND: ("foo__bar__in", "['one', 'two', 'three']"))>
```

### Using `Filter`:
```python
filters.parse(MultiValueDict({
    'bar__baz__in': ['nine', 'ten', 'eleven']
}))

# Results
filters.valid == True
filters.Q == <Q: (AND: ("bar__baz__in", ['nine', 'ten', 'eleven']))>
```

**Comment by bennullgraham** [comment]
Thanks for the detailed report! Yes, this is a bug. I came across this recently and it had been so long since writing the library that I got confused by it myself!

`CharFilter` currently uses Django's `forms.CharField` to convert the raw query input into a Python object before Filternaut works with it. With CharFilter this is basically a no-op since it runs `str()` in the input and the input is already a string, but if you use e.g. `DateTimeFilter` then raw input of '2017-01-01' is turned into a `datetime` instance, which is useful.

Anyway, this falls apart with multi-valued input as the conversion is applied to the raw input as-is; so CharFilter ends up running `str()` on `['one', 'two', 'three']` and you get the result described above.

If you're interested in making a patch, https://github.com/bennullgraham/filternaut/blob/master/filternaut/filters.py#L44 probably needs to special-case input which appears to be a list, tuple, etc. Otherwise, I'll fix when I get a chance!

**Comment by bennullgraham** [comment]
Turns out I got a chance fairly quickly :)

@jameswyse can you check that the `fix/multi-clean` branch (https://github.com/bennullgraham/filternaut/tree/fix/multi-clean) fixes or would appear to fix your case?

**Comment by jameswyse** [comment]
Thanks for the detailed response! (and then for fixing it!)

Your branch does appear to fix the problem I was seeing  :grin:

**Comment by bennullgraham** [comment]
Fixed in 0.0.9 which is now on PyPI. Thanks again for the ticket :+1:
